src/main/java/de/tum/in/www1/artemis/service/hestia/ExerciseHintService.java

```diff
@@ -215,9 +213,14 @@ private boolean isTaskSuccessfulInSubmission(ProgrammingExerciseTask task, Submi
         }
         var testCasesInTask = task.getTestCases();
         var feedbacks = result.getFeedbacks();
-        // the feedback can either not exist (e.g. for skipped test cases), or the attribute positive may not be set
-        return feedbacks.stream().filter(feedback -> feedback != null && testCasesInTask.stream().anyMatch(testCase -> Objects.equals(testCase.getTestName(), feedback.getText())))
-                .allMatch(feedback -> Boolean.TRUE.equals(feedback.isPositive()));
+        // a task is successful if feedback for all in the task linked tests exist and if these tests all passed.
+        var feedbackForTask = testCasesInTask.stream().map(testCase -> feedbacks.stream().filter(feedback -> testCase.equals(feedback.getTestCase())).findAny())
```

Comment:
IMO this is a bit hard to understand all at once, but since it didn't get worse compared to the previous version of this, I would be ok with it for this PR.